receiver/dockerstatsreceiver/metadata.yaml

```diff
@@ -628,3 +628,24 @@ metrics:
       aggregation: cumulative
     attributes:
       - interface
+
+  # Pids
+  container.pids.count:
+    enabled: false
+    description: "Number of pids in the container's cgroup."
+    extended_documentation: "It requires docker API 1.23 or higher and kernel version >= 4.3 with pids cgroup supported. [More docs](https://www.kernel.org/doc/Documentation/cgroup-v1/pids.txt)"
+    unit: "{pids}"
+    sum:
+      value_type: int
+      aggregation: cumulative
+      monotonic: false
+
+  container.pids.max:
```

Comment:
Is there a reason you've gone with "max" as the terminology here? Seeing as docker and cgroups report it as "limit" it might make sense for the metric name to be `container.pids.limit`. This is also consistent with other metrics in this component such as `container.memory.usage.limit`.

Reply:
I took it from [cgroups docs](https://www.kernel.org/doc/Documentation/cgroup-v1/pids.txt):

> In order to use the `pids` controller, set the maximum number of tasks in pids.max (this is not available in the root cgroup for obvious reasons). The number of processes currently in the cgroup is given by pids.current.

But, as you stated, it is reported as limit and that name would be consistent with other metrics and the conventions, so I'm changing it. Thanks again! 🙂

Reply:
Changed here: [653ad44](https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/21111/commits/653ad44e3a77e6d1673c5bf19757ac8dc2567008)